### PR TITLE
Initial work to deliver selected svelte components as web components

### PR DESCRIPTION
**Pull request by yauchow**
Added initial support to expose SagaFlow UI elements as web components.

Added route to deliver deliver the web components js file.

Converted SagaFlow.UI to new Javascript project system file, added it as a project reference to the main SagaFlow project. Removed SvelteKit implementation.

Added npm build scripts to build selected SagaFlow components as web components.

Added initial support to connect to mulitple Sagaflow servers via initialization with a Server Key.

Added initial web component Command Form, that will render a basic form using the Config metadata of the given command.

### CODE REVIEW

File: SagaFlow.UI/package.json [hunk 1]
```diff
@@ -6,15 +6,19 @@
...
+    "build:web-components": "vite build --config ./vite.webComponents.config.ts",
     "preview": "vite preview",
     "check": "svelte-check --tsconfig ./tsconfig.json"
   },
   "devDependencies": {
     "@sveltejs/vite-plugin-svelte": "^2.0.4",
     "@tsconfig/svelte": "^4.0.1",
     "@types/node": "^20.2.6",
+    "sass": "^1.77.4",
+    "scss": "^0.2.4",
```
Comment (rburnham-def) [inline SagaFlow.UI/package.json:18]: We shouldn't need scss. what are we using it for?

File: SagaFlow.UI/package.json [hunk 2]
```diff
@@ -6,15 +6,19 @@
...
     "svelte": "^3.58.0",
```
Comment (rburnham-def) [inline SagaFlow.UI/package.json:19]: We should upgrade to Svelte 4 to make use of the new Web Component improvements 

https://svelte.dev/docs/v4-migration-guide#custom-elements-with-svelte

File: SagaFlow.UI/src/lib/CommandForm.svelte [hunk 3]
```diff
@@ -0,0 +1,137 @@
+<svelte:options tag={"sf-command-form"}/>
+<script lang="ts">
+    import type {Command, Config} from "$lib/types";
+    import sagaFlow, {defaultSagaFlowServer} from "../state/SageFlowState";
+    import {onDestroy} from "svelte";
+    import ResourceSelector from "$lib/ResourceSelector.svelte";
+
+    export let serverkey: string = defaultSagaFlowServer;
```
Comment (rburnham-def) [inline SagaFlow.UI/src/lib/CommandForm.svelte:8]: Typescript Types are normally Pascal Case

File: SagaFlow.UI/src/lib/CommandForm.svelte [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    };
+
+    const sendCommand = () => {
+        if (sendingCommand) return;
+        
+        sendingCommand = true;
+
+        const command = Object.fromEntries(new FormData(form).entries());
+        
+        sagaFlow
```
Comment (rburnham-def) [inline SagaFlow.UI/src/lib/CommandForm.svelte:29]: Should make this method Async and await

File: SagaFlow.UI/src/lib/CommandForm.svelte [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    const resetCommand = () => sendingCommand = false;
+    
+    const showError = <TError>(error: TError) => {
+        console.error("Error: ", error);
+        
+        lastErrorMessage = `${error}`;
+    }
+
+    const unsubscribe = sagaFlow
```
Comment (rburnham-def) [inline SagaFlow.UI/src/lib/CommandForm.svelte:42]: Your should make sagaFlow a derived store so you can use the $sagaflow syntax and have it manage subscribe and unsubscribe.

**Review by rburnham-def — APPROVED** [review]
Looks good. Added a few suggestions

File: SagaFlow.UI/src/lib/CommandForm.svelte (hunk 5 above)
Comment (yauchow) [reply SagaFlow.UI/src/lib/CommandForm.svelte:42]: $ syntax is actually a sveltekit concept and not available for vanilla svelte.

Comment (rburnham-def) [reply SagaFlow.UI/src/lib/CommandForm.svelte:42]: No it's not, This is what i mean https://learn.svelte.dev/tutorial/auto-subscriptions

Then you can probably mix that with derived stores https://learn.svelte.dev/tutorial/derived-stores

File: SagaFlow.UI/package.json (hunk 1 above)
Comment (yauchow) [reply SagaFlow.UI/package.json]: fixed

File: SagaFlow.UI/package.json (hunk 2 above)
Comment (yauchow) [reply SagaFlow.UI/package.json]: Updated

File: SagaFlow.UI/src/lib/CommandForm.svelte (hunk 5 above)
Comment (yauchow) [reply SagaFlow.UI/src/lib/CommandForm.svelte]: fixed

File: SagaFlow.UI/src/lib/CommandForm.svelte (hunk 4 above)
Comment (yauchow) [reply SagaFlow.UI/src/lib/CommandForm.svelte]: fixed

File: SagaFlow.UI/src/state/SageFlowState.ts
Comment (tysonstolarski) [inline SagaFlow.UI/src/state/SageFlowState.ts:1]: typo in filename (Sage vs Saga)

**Review by tysonstolarski — CHANGES_REQUESTED** [review]
Amazing! Only thing I could find was the typo in filename.

**Review by rburnham-def — APPROVED** [review]

**Pull request merged**